EmailPiece.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:core';
+import 'package:enough_mail/enough_mail.dart';
+import 'package:intl/intl.dart';
+import 'package:summer2022/models/Digest.dart';
+
+class EmailPiece {
+  String EmailPieceID; // Unique ID of Image
+  String EmailID; // Unique ID of Email - made from Email contents hashed

```

Comment:
This isn't a comment that needs resolving, but just noting how we intend to use "EmailID": it shouldn't be a hash or anything we create, since the goal is be able to quickly retrieve an individual email again from the server. Thus, the ID needs to be based on data from our email provider (enough_mail for now). When a user navigates to view an individual MailPiece, we will use this ID to request the full email from the server again in order to pull the image for display to the user (since we aren't caching the actual image itself).

Reply:
I can work with that - however that still leaves the question of how to get a unique ID for each Email, as how at least this is being processed so far is in the Cache function - with MailPiece's ID being just EmailID+J iteration as a string with J iterating up with each image per that email. 

And my first thought is that having each individual user having a counter that iterates up with each email cahced, but I'd prefer not to make more work for everyone.